src/VisualStudio/IntegrationTest/TestUtilities/InProcess/TextViewWindow_InProc.cs

```diff
@@ -300,19 +312,23 @@ private IEnumerable<ISuggestedAction> GetLightBulbActions(ILightBulbBroker broke
                 actionSets = Array.Empty<SuggestedActionSet>();
             }
 
-            return SelectActions(actionSets);
+            return await SelectActionsAsync(actionSets);
         }
 
         public void ApplyLightBulbAction(string actionName, FixAllScope? fixAllScope, bool blockUntilComplete)
         {
             var lightBulbAction = GetLightBulbApplicationAction(actionName, fixAllScope);
-            if (blockUntilComplete)
+            var task = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                ExecuteOnActiveView(lightBulbAction);
-            }
-            else
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                var activeTextView = GetActiveTextView();
+                await lightBulbAction(activeTextView);
+            });
+
+            if (blockUntilComplete)
             {
-                BeginInvokeExecuteOnActiveView(lightBulbAction);
+                task.Join();
```

Comment:
Is there a way to add an assert here that we're not currently on the main thread?

Reply:
➡️ This is `JoinableTask.Join`; there is no caller affinity restriction.